src/vsphere_cpi/lib/cloud/vsphere/nsxt_provider.rb

```diff
@@ -118,26 +123,32 @@ def initialize(client, default_vif_type)
     def add_vm_to_nsgroups(vm, ns_groups)
       return if ns_groups.nil? || ns_groups.empty?
       return if nsxt_nics(vm).empty?
-
+      lports = logical_ports(vm)
       logger.info("Adding vm '#{vm.cid}' to NSGroups: #{ns_groups}")
       nsgroups = retrieve_nsgroups(ns_groups)
-
-      lports = logical_ports(vm)
       nsgroups.each do |nsgroup|
         logger.info("Adding LogicalPorts: #{lports.map(&:id)} to NSGroup '#{nsgroup.id}'")
-        grouping_obj_svc.add_or_remove_ns_group_expression(
-          nsgroup.id,
-          *to_simple_expressions(lports),
-          'ADD_MEMBERS'
-        )
+        Bosh::Retryable.new(
+            tries: 50,
+            on: [NSXTOptimisticUpdateError]
+        ).retryer do |i|
+          begin
```

Comment:
You don't need the `begin` here. Since Ruby 2.5 `do ... end` blocks have an implicit `begin ... end`. See https://blog.bigbinary.com/2017/10/24/ruby-2.5-allows-rescue-inside-do-end-blocks.html.